lib/bus.js

```diff
@@ -160,6 +161,38 @@ Bus.prototype.validate = validateAndCheckGateway;
 Bus.prototype.toString = Object
   .getOwnPropertyDescriptor(Device.prototype, 'address').get;
 
+Bus.prototype.scan = function scan(timeout, callback) {
+
+  // Establish an empty device list
+  const devices = [];
+
+  this.on("command", function(command) {
```

Comment:
You should listen here only for correspondent response to `0x000E` command.

Reply:
Thanks you for all your feedback. We are going through them and making changes accordingly. About this one: I can't seem to find any documentation on a response for this one but will do som debugging to see what I can find. Would you prefer if we included this in commands.js?

Reply:
You can use this library to monitor messages in bus. Just start following snippet with `DEBUG=smart-bus` env variable.

```js
var bus = new Bus('hdl://1.85@192.168.10.255:6000');

bus.on('listening', function() {
  bus.setBroadcast(true);
});
```

Or setup custom logging

```js
bus.on('command', function(command) {
  console.log('Sender:', command.sender.address);
  console.log('Code:', command.code);
  console.log('Data:', command.data);
});
```

Then start scan mode with official `HDL Buspro Setup Tool` and check what is happening. Devices will respond to `0x000E` with some command code.

If you know how convert response data then this command code should be included to `commands.js`.